goal_src/engine/ui/hud-classes.gc

```diff
@@ -7,10 +7,3015 @@
+                                                                       (t9-1
+                                                                        (the-as
+                                                                         hud-money-all
+                                                                         s5-0
+                                                                         )
+                                                                        *target*
+                                                                        'hud-money-all
+                                                                        (the-as
+                                                                         pointer
+                                                                         #x70004000
+                                                                         )
+                                                                        )
+                                                                       )
+                                                                      ((the-as
+                                                                        (function process function object object)
+                                                                        run-function-in-process
+                                                                        )
+                                                                       s5-0
+                                                                       hud-init-by-other
+                                                                       arg0
+                                                                       )
+                                                                      (->
+                                                                       s5-0
+                                                                       ppointer
+                                                                       )
+                                                                      )
+                                      )
+     )
+    )
+   )
+  0
   )
 
-;; decomp begins
+(defbehavior convert-to-hud-object hud ((arg0 process-drawable) (arg1 hud))
+  (when arg1
+   (logclear! (-> arg0 mask) (process-mask actor-pause))
+   (let ((s4-0 (new 'stack-no-clear 'vector)))
+    (transform-point-vector-scale! s4-0 (-> arg0 root trans))
+    (set! (-> arg0 root trans x) (+ -2048.0 (-> s4-0 x)))
+    (set! (-> arg0 root trans y) (* 2.0 (+ -2048.0 (-> s4-0 y))))
+    )
+   (set! (-> arg0 root trans z) 9999999.0)
+   (set! (-> arg0 root trans w) 1.0)
+   (set-vector!
+    (-> arg0 root scale)
+    (inc-value arg1)
+    (dec-value arg1)
+    (inc-value arg1)
+    1.0
+    )
+   (set! (-> arg0 draw dma-add-func) dma-add-process-drawable-hud)
+   enter-state
+   (process->handle arg1)
+   (go hud-collecting)
```

Comment:
I think the `hud-collecting` state should take a single argument (`handle`)

Reply:
you're right, it does. However, doing so causes the handler in `hud` to crash.  Probably something that needs to be looked into (file isn't added yet so no harm in adding the arg now)